onnxruntime/core/providers/op_kernel_type_control.h

```diff
@@ -121,6 +121,29 @@ struct EnabledTypes {
   using types = boost::mp11::mp_apply<boost::mp11::mp_set_intersection, TypeSetsToConsider>;
 };
 
+/**
+* Check if the type list contains the specified type.
+*/
+template <typename TypeList, typename T>
+constexpr bool HasType() {
+  return std::is_same<boost::mp11::mp_set_contains<TypeList, T>, boost::mp11::mp_true>::value;
+}
+
+template <typename T>
+using SizeOfT = boost::mp11::mp_size_t<sizeof(T)>;
+
+/**
+* Check if the type list contains a type with the same size as T.
+* 
+* @remarks e.g. will return true if T is int32_t and the list contains any 4 byte type (i.e. sizeof(int32_t)) 
+*               such as int32_t, uint32_t or float.
+*/
+template <typename TypeList, typename T>
+constexpr bool HasTypeWithSameSize() {
+  using EnabledTypeSizes = boost::mp11::mp_unique<boost::mp11::mp_transform<SizeOfT, TypeList>>;
```

Comment:
>mp_unique [](start = 40, length = 9)

cool, i didn't know about this when i added EnabledTypes::MakeSet above. mind also replacing that with mp_unique?